lib/SensorFusion/src/ExtendedKalmanFilter.h

```diff
@@ -95,17 +95,20 @@ class ExtendedKalmanFilter : public IKalmanFilter
     void init(KalmanParameter& initialParameter) final;
 
     /**
-     * Prediction of the covariance and the state of the Extended Kalman Filter.
+     * Prediction of the covariance and the state of the Kalman Filter.
+     * 
      * @param[in] timeStep Measured Time Step in ms.
+     * @param[in] kalmanParameter   Input Parameters for the Kalman Filter
      */
-    void predictionStep(const uint16_t timeStep) final;
+    virtual void predictionStep(const uint16_t timeStep, KalmanParameter& kalmanParameter) final;

```

Comment:
The parameters can change between each step, or why do you need to pass them each time?

Reply:
Yes, these are the Measurement Values from the IMU and the Odometry.